tests/integration/test_consume_flow.py

```diff
@@ -0,0 +1,159 @@
+        publisher_wallet,
+    )
+    tx_receipt = web3.eth.wait_for_transaction_receipt(tx)
+    registered_event = erc721_factory.get_event_log(
+        ERC721FactoryContract.EVENT_NFT_CREATED,
+        tx_receipt.blockNumber,
+        web3.eth.block_number,
+        None,
+    )
+
+    assert registered_event[0].event == "NFTCreated"
+    assert registered_event[0].args.admin == publisher_wallet.address
+    erc721_address = registered_event[0].args.newTokenAddress
+
+    data_provider = DataServiceProvider
+    asset = OceanAssets(config, web3, data_provider)
+    metadata = {
+        "created": "2020-11-15T12:27:48Z",
+        "updated": "2021-05-17T21:58:02Z",
+        "description": "Sample description",
+        "name": "Sample asset",
+        "type": "dataset",
+        "author": "OPF",
+        "license": "https://market.oceanprotocol.com/terms",
+    }
+    file_url = "https://raw.githubusercontent.com/tbertinmahieux/MSongsDB/master/Tasks_Demos/CoverSongs/shs_dataset_test.txt"
+    file_dict = {"type": "url", "url": file_url, "method": "GET"}
+    file = FilesTypeFactory(file_dict)
+    files = [file]
+
+    # Encrypt file objects
+    encrypt_response = data_provider.encrypt(
+        files, "http://172.15.0.4:8030/api/services/encrypt"
+    )
+    encrypted_files = encrypt_response.content.decode("utf-8")
+
+    # Set ERC20 Data
+    erc20_data = ErcCreateData(
+        template_index=1,
+        strings=["Datatoken 1", "DT1"],
+        addresses=[
+            publisher_wallet.address,
+            publisher_wallet.address,
+            ZERO_ADDRESS,
+            get_address_of_type(config, "Ocean"),
+        ],
+        uints=[web3.toWei("100", "ether"), 0],
+        bytess=[b""],
+    )
+
+    # Publish a plain asset with one data token on chain
+    ddo = asset.create(
+        metadata=metadata,
+        publisher_wallet=publisher_wallet,
+        encrypted_files=encrypted_files,
+        erc721_address=erc721_address,
+        erc20_tokens_data=[erc20_data],
+    )
+
+    assert ddo, "The asset is not created."
+    assert ddo.nft["name"] == "NFT1"
+    assert ddo.nft["symbol"] == "NFTSYMBOL"
+    assert ddo.nft["address"] == erc721_address
+    assert ddo.nft["owner"] == publisher_wallet.address
+    assert ddo.datatokens[0]["name"] == "Datatoken 1"
+    assert ddo.datatokens[0]["symbol"] == "DT1"
+
+    service = ddo.get_service(ServiceTypes.ASSET_ACCESS)
+    erc20_token = ERC20Token(web3, ddo.datatokens[0]["address"])
+
+    # Mint 50 ERC20 tokens in consumer wallet from publisher. Max cap = 100
+    erc20_token.mint(
+        account_address=consumer_wallet.address,
+        value=to_wei(50),
+        from_wallet=publisher_wallet,
+    )
+
+    # Initialize service
+    response = data_provider.initialize(
+        did=ddo.did,
+        service_id=service.id,
+        consumer_address=consumer_wallet.address,
+        service_endpoint=data_provider.build_initialize_endpoint(config.provider_url)[
```

Comment:
please assign the service_endpoint separately, using a _, variable = build() syntax. It is clearer if we switch the tuple count at some point. The [1] selector can apply to a list too, and it's a bit unclear.